File: Java/src/main/java/QuestionBank/Solution201_300/Solution219.java
package QuestionBank.Solution201_300;

import java.util.HashMap;

/**
 * @author windows
 */
public class Solution219 {
    public boolean containsNearbyDuplicate(int[] nums, int k) {
        HashMap<Integer, Integer> map = new HashMap<>((int) (nums.length / 0.75 + 1));
        for (int i = 0; i < nums.length; i++) {
            if (!map.containsKey(nums[i])) {
                map.put(nums[i], i);
            }else{
                int index = map.get(nums[i]);
                if (Math.abs(index - i) <= k){
                    return true;
                }else{
                    map.put(nums[i], i);
                }
            }
        }
        return false;
    }

    public static void main(String[] args) {
        Solution219 solution219 = new Solution219();
    }
}

File: C++/src/Solution201_300/Solution219.cpp
//
// Created by windows on 2023/3/1.
//

#include "Solution219.h"
#include <iostream>
#include <vector>
#include <unordered_set>
#include <unordered_map>
#include <string>
using namespace std;
#include <unordered_map>

bool Solution219::containsNearbyDuplicate(vector<int> &nums, int k) {
    unordered_map<int, int> hashtable;
    for (int i = 0; i < nums.size(); ++i) {
        if (hashtable.count(nums[i]) && i - hashtable[nums[i]] <= k){
            return true;
        }
        hashtable[nums[i]] = i;
    }
    return false;
}
